### PR TITLE
Add revoke access to Manila

### PR DESCRIPTION
**Pull request by gman0**
For #1076

Links to the line numbers/files in the OpenStack source code that support the
code in this PR:

- revoke access request https://github.com/openstack/manila/blob/master/api-ref/source/samples/share-actions-revoke-access-request.json
- removing an access rule from DB https://github.com/openstack/manila/blob/master/manila/share/access.py#L224

### CODE REVIEW

**Comment by gman0** [comment]
cc @jtopjian

File: openstack/sharedfilesystems/v2/shares/testing/fixtures.go [hunk 1]
```diff
@@ -198,6 +198,25 @@ func MockGrantAccessResponse(t *testing.T) {
...
+func MockRevokeAccessResponse(t *testing.T) {
+	th.Mux.HandleFunc(shareEndpoint+"/"+shareID+"/action", func(w http.ResponseWriter, r *http.Request) {
+		th.TestMethod(t, r, "POST")
+		th.TestHeader(t, r, "X-Auth-Token", fake.TokenID)
+		th.TestHeader(t, r, "Content-Type", "application/json")
+		th.TestHeader(t, r, "Accept", "application/json")
+		th.TestJSONRequest(t, r, revokeAccessRequest)
+		w.Header().Add("Content-Type", "application/json")
+		w.WriteHeader(http.StatusOK)
+		fmt.Fprintf(w, grantAccessResponse)
```
Comment (jtopjian) [inline openstack/sharedfilesystems/v2/shares/testing/fixtures.go]: I don't think this API call actually returns anything, correct? If so, then let's remove this line. Maybe change `StatusOK` to `StatusAccepted` to emulate the 202 response, too. I'm OK with leaving 200 as a valid `OkCode` in addition to the 202.

Comment (gman0) [reply openstack/sharedfilesystems/v2/shares/testing/fixtures.go]: Ah yes, absolutely!

**Comment by gman0** [comment]
Thanks!

**Pull request merged**